File: backend/src/main/java/tom/projects/monkeylog/dto/workout/ExerciseRowResponse.java
package tom.projects.monkeylog.dto.workout;

import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.util.List;
import java.util.UUID;

@Data
public class ExerciseRowResponse {
    @NotNull
    private UUID id;
    @NotNull
    private List<ExerciseRowFieldResponse> exerciseRowFields;
    @NotNull
    private Boolean isLifted;
}

File: backend/src/main/kotlin/tom/com/monkeylog/dto/workout/ExerciseRowResponse.kt
package tom.com.monkeylog.dto.workout

import java.util.*

data class ExerciseRowResponse(
    val id: UUID,
    val lifted: Boolean,
    val weight: Double?,
    val distance: Double?,
    val time: Int?,
    val reps: Int?,
    val rpe: Int?
)
